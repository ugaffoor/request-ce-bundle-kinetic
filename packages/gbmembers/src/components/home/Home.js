import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  compose,
  withState,
  lifecycle,
  withHandlers,
  withProps,
} from 'recompose';
import { actions } from '../../redux/modules/members';
import { actions as appActions } from '../../redux/modules/memberApp';
import $ from 'jquery';
import moment from 'moment';
import PropTypes from 'prop-types';
import 'bootstrap/scss/bootstrap.scss';
import { StatusMessagesContainer } from '../StatusMessages';
import { actions as errorActions } from '../../redux/modules/errors';
import { Utils } from 'common';
import { BamboraOverdues } from './BamboraOverdues';
import { StripeOverdues } from './StripeOverdues';
import { DemographicChart } from './Demographic';
import { ProgramsChart } from './Programs';
import { Statistics } from './Statistics';
import { LeadsOriginChart } from './LeadsOrigin';
import { AttendancePerDay } from './AttendancePerDay';
import { Finances } from './Finances';
import { actions as leadsActions } from '../../redux/modules/leads';
import { actions as attendanceActions } from '../../redux/modules/attendance';
import { actions as monthlyStatisticsActions } from '../../redux/modules/monthlyStatistics';
import { actions as classActions } from '../../redux/modules/classes';
import { getAttributeValue } from '../../lib/react-kinops-components/src/utils';

const mapStateToProps = state => ({
  pathname: state.router.location.pathname,
  memberItem: state.member.members.currentMember,
  membersLoading: state.member.members.membersLoading,
  allMembers: state.member.members.allMembers,
  billingPayments: state.member.members.billingPayments,
  billingPaymentsLoading: state.member.members.billingPaymentsLoading,
  billingCompany: state.member.app.billingCompany,
  programs: state.member.app.programs,
  profile: state.member.app.profile,
  space: state.member.app.space,
  leadsByDate: state.member.leads.leadsByDate,
  leadsLoading: state.member.leads.leadsLoading,
  leadsByDateLoading: state.member.leads.leadsByDateLoading,
  fetchingAttendancesByDate: state.member.attendance.fetchingAttendancesByDate,
  attendancesByDate: state.member.attendance.attendancesByDate,
  classSchedules: state.member.classes.classSchedules,
  fetchingClassSchedules: state.member.classes.fetchingClassSchedules,
  monthlyStatistics: state.member.monthlyStatistics.monthlyStatistics,
  monthlyStatisticsLoading:
    state.member.monthlyStatistics.monthlyStatisticsLoading,
  FAILEDpaymentHistory: state.member.members.FAILEDpaymentHistory,
  FAILEDpaymentHistoryLoading: state.member.members.FAILEDpaymentHistoryLoading,
  SUCCESSFULpaymentHistory: state.member.members.SUCCESSFULpaymentHistory,
  SUCCESSFULpaymentHistoryLoading:
    state.member.members.SUCCESSFULpaymentHistoryLoading,
});

const mapDispatchToProps = {
  fetchCurrentMember: actions.fetchCurrentMember,
  fetchBillingPayments: actions.fetchBillingPayments,
  fetchLeadsByDate: leadsActions.fetchLeadsByDate,
  fetchAttendancesByDate: attendanceActions.fetchAttendancesByDate,
  fetchClassSchedules: classActions.fetchClassSchedules,
  setBillingPayments: actions.setBillingPayments,
  addNotification: errorActions.addNotification,
  setSystemError: errorActions.setSystemError,
  fetchMonthlyStatistics: monthlyStatisticsActions.fetchMonthlyStatistics,
  setSidebarDisplayType: appActions.setSidebarDisplayType,
  fetchPaymentHistory: actions.fetchPaymentHistory,
  setPaymentHistory: actions.setPaymentHistory,
};

export const HomeView = ({
  memberItem,
  allMembers,
  membersLoading,
  billingPayments,
  getBillingPayments,
  billingPaymentsLoading,
  billingCompany,
  programs,
  profile,
  leadsByDate,
  fetchLeadsByDate,
  leadsByDateLoading,
  attendancesByDate,
  fetchAttendancesByDate,
  fetchingAttendancesByDate,
  fetchClassSchedules,
  classSchedules,
  fetchingClassSchedules,
  monthlyStatistics,
  fetchMonthlyStatistics,
  fetchingMonthlyStatistics,
  datesChanged,
  setFromDate,
  setToDate,
  fromDate,
  toDate,
  setIsAssigning,
  space,
  currency,
  locale,
  getFailedPayments,
  FAILEDpaymentHistory,
  FAILEDpaymentHistoryLoading,
  getSuccessfulPayments,
  SUCCESSFULpaymentHistory,
  SUCCESSFULpaymentHistoryLoading,
}) => (
  <div className="dashboard">
    <StatusMessagesContainer />
    <Statistics
      setIsAssigning={setIsAssigning}
      setFromDate={setFromDate}
      setToDate={setToDate}
      fromDate={fromDate}
      toDate={toDate}
      datesChanged={datesChanged}
      leadsByDate={leadsByDate}
      fetchLeadsByDate={fetchLeadsByDate}
      leadsByDateLoading={leadsByDateLoading}
      allMembers={allMembers}
      membersLoading={membersLoading}
      profile={profile}
      space={space}
    />
    {getAttributeValue(space, 'Billing Company') === 'Bambora' && (
      <div className="homeOverdues">
        {!membersLoading && (
          <div>
            <BamboraOverdues
              allMembers={allMembers}
              getFailedPayments={getFailedPayments}
              paymentHistory={FAILEDpaymentHistory}
              FAILEDpaymentHistoryLoading={FAILEDpaymentHistoryLoading}
              getSuccessfulPayments={getSuccessfulPayments}
              successfulPaymentHistory={SUCCESSFULpaymentHistory}
              SUCCESSFULpaymentHistoryLoading={SUCCESSFULpaymentHistoryLoading}
              space={space}
              locale={locale}
              profile={profile}
            />
          </div>
        )}
      </div>
    )}
    {getAttributeValue(space, 'Billing Company') === 'Stripe' && (
      <div className="homeOverdues">
        {!membersLoading && (
          <div>
            <StripeOverdues
              allMembers={allMembers}
              getFailedPayments={getFailedPayments}
              paymentHistory={FAILEDpaymentHistory}
              FAILEDpaymentHistoryLoading={FAILEDpaymentHistoryLoading}
              getSuccessfulPayments={getSuccessfulPayments}
              successfulPaymentHistory={SUCCESSFULpaymentHistory}
              SUCCESSFULpaymentHistoryLoading={SUCCESSFULpaymentHistoryLoading}
              space={space}
              locale={locale}
              profile={profile}
            />
          </div>
        )}
      </div>
    )}
    <div className="charts">
      <div className="chart2Column">
        <div className="col1 chart5">
          <ProgramsChart allMembers={allMembers} programs={programs} />
        </div>
        <div className="col2 chart7">
          <AttendancePerDay
            fromDate={fromDate}
            toDate={toDate}
            attendancesByDate={attendancesByDate}
            fetchAttendancesByDate={fetchAttendancesByDate}
            fetchingAttendancesByDate={fetchingAttendancesByDate}
            fetchClassSchedules={fetchClassSchedules}
            classSchedules={classSchedules}
            fetchingClassSchedules={fetchingClassSchedules}
            allMembers={allMembers}
          />
        </div>
      </div>
      <div className="chart2Column">
        <div className="col1 chart4">
          <DemographicChart allMembers={allMembers} space={space} />
        </div>
        <div className="col2 chart6">
          <LeadsOriginChart
            fromDate={fromDate}
            toDate={toDate}
            leadsByDate={leadsByDate}
            fetchLeadsByDate={fetchLeadsByDate}
            leadsByDateLoading={leadsByDateLoading}
          />
          {/*      <KidsChart allMembers={allMembers} /> */}
        </div>
      </div>
      {!Utils.isMemberOf(profile, 'Billing') ? (
        <div />
      ) : (
        <div className="chart1Column">
          <div className="col">
            <Finances
              monthlyStatistics={monthlyStatistics}
              fetchMonthlyStatistics={fetchMonthlyStatistics}
              fetchingMonthlyStatistics={fetchingMonthlyStatistics}
              currency={currency}
              locale={locale}
            />
          </div>
        </div>
      )}
    </div>
  </div>
);

export const HomeContainer = compose(
  connect(mapStateToProps, mapDispatchToProps),
  withProps(({ memberItem, space, profile }) => {
    return {};
  }),
  withState('isAssigning', 'setIsAssigning', false),
  withState('fromDate', 'setFromDate', moment().subtract(7, 'days')),
  withState('toDate', 'setToDate', moment()),
  withHandlers({
    getFailedPayments: ({
      fetchPaymentHistory,
      setPaymentHistory,
      addNotification,
      setSystemError,
    }) => () => {
      fetchPaymentHistory({
        paymentType: 'FAILED',
        paymentMethod: 'ALL',
        paymentSource: 'ALL',
        dateField: 'PAYMENT',
        dateFrom: moment()
          .subtract(6, 'month')
          .format('YYYY-MM-DD'),
        dateTo: moment().format('YYYY-MM-DD'),
        setPaymentHistory: setPaymentHistory,
        internalPaymentType: 'client_failed',
        addNotification: addNotification,
        setSystemError: setSystemError,
      });
    },
    getSuccessfulPayments: ({
      fetchPaymentHistory,
      setPaymentHistory,
      addNotification,
      setSystemError,
    }) => () => {
      fetchPaymentHistory({
        paymentType: 'SUCCESSFUL',
        paymentMethod: 'ALL',
        paymentSource: 'ALL',
        dateField: 'PAYMENT',
        dateFrom: moment()
          .subtract(1, 'month')
          .format('YYYY-MM-DD'),
        dateTo: moment().format('YYYY-MM-DD'),
        setPaymentHistory: setPaymentHistory,
        internalPaymentType: 'client_successful',
        addNotification: addNotification,
        setSystemError: setSystemError,
      });
    },
    getBillingPayments: ({
      memberItem,
      fetchBillingPayments,
      setBillingPayments,
      addNotification,
      setSystemError,
    }) => month => {
      let startDate, endDate;
      if (month === 'current_month') {
        startDate = moment
          .utc()
          .startOf('month')
          .format('YYYY-MM-DD');
        endDate = moment.utc().format('YYYY-MM-DD');
      } else if (month === 'previous_month') {
        startDate = moment
          .utc()
          .subtract(2, 'months')
          .startOf('month')
          .format('YYYY-MM-DD');
        endDate = moment
          .utc()
          .subtract(2, 'months')
          .endOf('month')
          .format('YYYY-MM-DD');
      }
      fetchBillingPayments({
        paymentType: 'SUCCESSFUL',
        paymentMethod: 'ALL',
        paymentSource: 'ALL',
        dateField: 'PAYMENT',
        dateFrom: startDate,
        dateTo: endDate,
        setBillingPayments: setBillingPayments,
        internalPaymentType: 'client_successful',
        addNotification: addNotification,
        setSystemError: setSystemError,
      });
    },
    datesChanged: () => (setFromDate, setToDate, fromDate, toDate) => {
      console.log('datesChanged');
      setFromDate(fromDate);
      setToDate(toDate);
    },
  }),
  lifecycle({
    UNSAFE_componentWillMount() {
      if (
        Utils.isMemberOf(this.props.profile, 'Billing') &&
        getAttributeValue(this.props.space, 'Billing Company') !== 'No Billing'
      ) {
        if (
          !this.props.billingPayments ||
          this.props.billingPayments.length <= 0
        ) {
          this.props.getBillingPayments('current_month');
        }
      }
      //      this.props.fetchLeadsByDate();

      let currency = getAttributeValue(this.props.space, 'Currency');
      if (currency === undefined) currency = 'USD';

      this.locale =
        this.props.profile.preferredLocale === null
          ? this.props.space.defaultLocale
          : this.props.profile.preferredLocale;

      this.setState({
        currency: currency,
        locale: this.locale,
      });
    },
    UNSAFE_componentWillReceiveProps(nextProps) {
      this.props.setSidebarDisplayType('members');
      $('.content')
        .parent('div')[0]
        .scrollIntoView(true);
    },
    componentDidMount() {},
    UNSAFE_componentWillUnmount() {},
  }),
)(HomeView);

function renderText(child, x, y, rotate, stroke, key) {
  if (child && child.content) {
    return (
      <text
        key={key}
        x={x}
        y={y}
        transform={`rotate(${rotate})`}
        textAnchor="middle"
        stroke={stroke}
        {...child.props}
      >
        {child.content}
      </text>
    );
  }

  return (
    <text
      key={key}
      x={x}
      y={y}
      transform={`rotate(${rotate})`}
      textAnchor="middle"
      stroke={stroke}
    >
      {child}
    </text>
  );
}

export default function AxisLabel({
  axisType,
  x,
  y,
  width,
  height,
  stroke,
  children,
}) {
  const isVert = axisType === 'yAxis';
  const cx = isVert ? x : x + width / 2;
  const cy = isVert ? height / 2 + y : y + height + 20;
  const rot = isVert ? `270 ${cx} ${cy}` : 0;
  const lineHeight = 20;

  if (children.length > 1 && children.map) {
    return (
      <g>
        {children.map((child, index) =>
          renderText(child, cx, cy + index * lineHeight, rot, stroke, index),
        )}
      </g>
    );
  }

  return renderText(children, cx, cy, rot, stroke);
}

AxisLabel.propTypes = {
  axisType: PropTypes.oneOf(['yAxis', 'xAxis']),
  x: PropTypes.number,
  y: PropTypes.number,
  width: PropTypes.number,
  height: PropTypes.number,
  stroke: PropTypes.string,
  children: PropTypes.any,
};
