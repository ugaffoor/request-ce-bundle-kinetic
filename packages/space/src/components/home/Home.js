import React from 'react';
import { Link } from 'react-router-dom';
import { compose, lifecycle, withHandlers } from 'recompose';
import { connect } from 'react-redux';
import { GroupDivider, PageTitle } from 'common';
import { selectDiscussionsEnabled } from 'common/src/redux/modules/common';
import {
  actions,
  selectIsMoreDiscussions,
  selectGroupedDiscussions,
} from '../../redux/modules/spaceApp';
import { actions as teamListActions } from '../../redux/modules/teamList';
import { CreateDiscussionModal } from './CreateDiscussionModal';
import { Discussion } from './Discussion';
import wallyMissingImage from 'common/src/assets/images/wally-missing.svg';
import { bundle } from 'react-kinetic-core';
import { I18n } from '../../../../app/src/I18nProvider';

const HomeComponent = ({
  spaceName,
  kapps,
  teams,
  discussionsEnabled,
  discussionGroups,
  discussionsError,
  discussionsLoading,
  discussionsSearchTerm,
  discussionsSearchInputValue,
  discussionServerUrl,
  isMoreDiscussions,
  me,
  handleCreateDiscussionButtonClick,
  handleDiscussionSearchInputChange,
  handleDiscussionSearchInputSubmit,
  handleLoadMoreButtonClick,
  handleHomeLinkClick,
}) => (
  <div className="page-container page-container--space-home">
    <PageTitle parts={['Home']} />
    <CreateDiscussionModal />
    <div className="page-panel page-panel--space-home">
      <h4 className="space-home-title">
        <I18n>Welcome to kinops for</I18n> {spaceName}
      </h4>
      {discussionsEnabled ? (
        <div className="page-title">
          <div className="page-title__wrapper">
            <h3>
              <Link onClick={handleHomeLinkClick} to="/">
                <I18n>home</I18n>
              </Link>{' '}
              /{' '}
              {discussionsSearchTerm !== '' ? <I18n>search results</I18n> : ''}
            </h3>
            <h1>
              {discussionsSearchTerm !== '' ? (
                `${discussionsSearchTerm}`
              ) : (
                <I18n>Recent Discussions</I18n>
              )}
            </h1>
          </div>
          <div className="search-form--discussion">
            <div className="search-box search-box--discussion">
              <form
                onSubmit={handleDiscussionSearchInputSubmit}
                className="search-box__form"
              >
                <div className="input-group">
                  <I18n
                    render={translate => (
                      <input
                        type="text"
                        placeholder={translate('Search discussions')}
                        onChange={handleDiscussionSearchInputChange}
                        className="form-control"
                        value={discussionsSearchInputValue}
                      />
                    )}
                  />
                  <div className="input-group-append">
                    <button className="btn" type="submit">
                      <span className="fa fa-search" />
                    </button>
                  </div>
                </div>
              </form>
            </div>
            <button
              onClick={handleCreateDiscussionButtonClick}
              className="btn btn-secondary"
            >
              <I18n>New Discussion</I18n>
            </button>
          </div>
        </div>
      ) : (
        <div className="empty-state empty-state--wally">
          <img src={wallyMissingImage} alt="Missing Wally" />
        </div>
      )}

      {discussionsEnabled &&
        !discussionsError &&
        !discussionsLoading &&
        discussionGroups.size > 0 && (
          <div className="recent-discussions-wrapper kinops-discussions">
            {discussionGroups
              .map((discussions, dateGroup) => (
                <div className="messages" key={dateGroup}>
                  <GroupDivider value={dateGroup} className="my-4" />
                  {discussions.map(discussion => (
                    <Discussion
                      key={discussion.id}
                      discussion={discussion}
                      teams={teams}
                      me={me}
                      discussionServerUrl={discussionServerUrl}
                    />
                  ))}
                </div>
              ))
              .toList()}
            {isMoreDiscussions && (
              <div className="recent-discussion-actions">
                <button
                  className="btn btn-primary"
                  onClick={handleLoadMoreButtonClick}
                >
                  <I18n>Load More</I18n>
                </button>
              </div>
            )}
          </div>
        )}
      {!discussionsError && !discussionsLoading && discussionGroups.size === 0 && (
        <div className="empty-state empty-state--wally">
          <h5>
            <I18n>No discussions found</I18n>
          </h5>
          <img src={wallyMissingImage} alt="Missing Wally" />
          <h6>
            <I18n>You are not involved in any discussions!</I18n>
          </h6>
        </div>
      )}
    </div>
  </div>
);

export const mapStateToProps = state => ({
  spaceName: state.app.space.name,
  discussionGroups: selectGroupedDiscussions(state),
  discussionsError: state.space.spaceApp.discussionsError,
  discussionsLoading: state.space.spaceApp.discussionsLoading,
  discussionsLimit: state.space.spaceApp.discussionsLimit,
  discussionsOffset: state.space.spaceApp.discussionsOffset,
  discussionsSearchInputValue: state.space.spaceApp.discussionsSearchInputValue,
  discussionsSearchTerm: state.space.spaceApp.discussionsSearchTerm,
  discussionServerUrl: `${bundle.spaceLocation()}/kinetic-response`,
  isMoreDiscussions: selectIsMoreDiscussions(state),
  discussionsEnabled: selectDiscussionsEnabled(state),
  me: state.app.profile,
  teams: state.space.teamList.data,
});

export const mapDispatchToProps = {
  fetchDiscussions: actions.fetchDiscussions,
  fetchTeams: teamListActions.fetchTeams,
  searchDiscussions: actions.searchDiscussions,
  setCreateDiscussionModalOpen: actions.setCreateDiscussionModalOpen,
  setDiscussionsLimit: actions.setDiscussionsLimit,
  setDiscussionsOffset: actions.setDiscussionsOffset,
  setDiscussionsSearchInputValue: actions.setDiscussionsSearchInputValue,
  setDiscussionsSearchTerm: actions.setDiscussionsSearchTerm,
};

export const Home = compose(
  connect(mapStateToProps, mapDispatchToProps),
  withHandlers({
    handleCreateDiscussionButtonClick: props => event =>
      props.setCreateDiscussionModalOpen(true),
    handleDiscussionSearchInputChange: props => event =>
      props.setDiscussionsSearchInputValue(event.target.value),
    handleDiscussionSearchInputSubmit: props => event => {
      event.preventDefault();
      props.setDiscussionsLimit(10);
      props.setDiscussionsOffset(0);
      props.searchDiscussions();
      props.setDiscussionsSearchTerm(props.discussionsSearchInputValue);
    },
    handleHomeLinkClick: props => event => {
      event.preventDefault();
      props.setDiscussionsSearchTerm('');
      props.setDiscussionsSearchInputValue('');
      props.searchDiscussions();
    },
    handleLoadMoreButtonClick: ({
      fetchDiscussions,
      discussionsLimit,
      setDiscussionsLimit,
    }) => event => {
      setDiscussionsLimit(discussionsLimit + 10);
      fetchDiscussions();
    },
  }),
  lifecycle({
    componentWillMount() {
      this.props.fetchTeams();
      this.props.discussionsEnabled && this.props.fetchDiscussions();
    },
  }),
)(HomeComponent);
