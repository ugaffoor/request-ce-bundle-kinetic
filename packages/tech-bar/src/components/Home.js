import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';
import {
  compose,
  lifecycle,
  withHandlers,
  withProps,
  withState,
} from 'recompose';
import {
  Modal,
  ModalBody,
  Dropdown,
  DropdownToggle,
  DropdownMenu,
} from 'reactstrap';
import {
  KappLink as Link,
  PageTitle,
  selectCurrentKapp,
  Moment,
  Constants,
  Utils,
} from 'common';
import { actions } from '../redux/modules/appointments';
import {
  SESSION_ITEM_USER_LOCATION,
  SESSION_ITEM_CURRENT_TECH_BAR,
  mapTechBarsForDistance,
  sortTechBarsByDistance,
} from '../redux/modules/techBarApp';
import moment from 'moment';
import { I18n } from '../../../app/src/I18nProvider';
import { DATE_FORMAT, TIME_FORMAT } from '../App';
import heroImage from '../assets/images/tech-bar-hero.jpg';

export const HomeComponent = ({
  kapp,
  techBars,
  loadingUpcoming,
  upcomingErrors,
  upcomingAppointments,
  modalOpen,
  setModalOpen,
  currentTechBar,
  selectCurrentTechBar,
  userLocation,
  getUserLocation,
  openDropdown,
  toggleDropdown,
  hasTechBarDisplayRole,
}) => {
  const selectedTechBar = currentTechBar || techBars.get(0);
  return (
    <Fragment>
      <PageTitle parts={[]} />
      <div
        className="hero--tech-bar"
        style={{ backgroundImage: `url(${heroImage})` }}
      >
        <div className="container">
          <div className="hero-welcome">
            <div className="title">
              <I18n>Welcome to {kapp ? kapp.name : 'Tech Bar'}</I18n>
            </div>
            <div className="subtitle">
              <I18n>
                We deliver in-person service that gets people back on track.
              </I18n>
            </div>
          </div>
          <div className="hero-card">
            <div className="card card--tech-bar-loc">
              <div className="card-body">
                <span className="icon fa fa-map-marker" />
                <div className="content">
                  <span className="title">
                    <I18n>{selectedTechBar.values['Name']}</I18n>
                    {techBars.size > 1 && (
                      <button
                        className="btn btn-inverse btn-sm"
                        onClick={() => setModalOpen(true)}
                      >
                        <I18n>Change</I18n>
                      </button>
                    )}
                  </span>
                  {selectedTechBar.values['Location'] && (
                    <div className="subtitle">
                      <I18n>{selectedTechBar.values['Location']}</I18n>
                    </div>
                  )}
                  <div className="details">
                    {selectedTechBar.values['Description'] && (
                      <p>
                        <I18n>{selectedTechBar.values['Description']}</I18n>
                      </p>
                    )}
                    {selectedTechBar.values['Details'] && (
                      <p>
                        <I18n>{selectedTechBar.values['Details']}</I18n>
                      </p>
                    )}
                  </div>
                </div>
                {/*hasTechBarDisplayRole(selectedTechBar.values['Name']) && (
                  <Dropdown
                    toggle={toggleDropdown(selectedTechBar.id)}
                    isOpen={openDropdown === selectedTechBar.id}
                  >
                    <DropdownToggle color="link" className="btn-sm">
                      <span className="fa fa-ellipsis-v fa-2x" />
                    </DropdownToggle>
                    <DropdownMenu right>
                      <Link
                        to={`/display/${selectedTechBar.values['Id']}/checkin`}
                        className="dropdown-item"
                        target="_blank"
                      >
                        <span className="fa fa-fw fa-external-link mr-2" />
                        <span>
                          <I18n>Check In</I18n>
                        </span>
                      </Link>
                      <Link
                        to={`/display/${selectedTechBar.values['Id']}/feedback`}
                        className="dropdown-item"
                        target="_blank"
                      >
                        <span className="fa fa-external-link fa-fw mr-2" />
                        <span>
                          <I18n>Feedback</I18n>
                        </span>
                      </Link>
                      <Link
                        to={`/display/${
                          selectedTechBar.values['Id']
                        }/checkin?crosslink`}
                        className="dropdown-item"
                        target="_blank"
                      >
                        <span className="fa fa-external-link fa-fw mr-2" />
                        <span>
                          <I18n>Check In</I18n> / <I18n>Feedback</I18n>
                        </span>
                      </Link>
                      <Link
                        to={`/display/${selectedTechBar.values['Id']}/overhead`}
                        className="dropdown-item"
                        target="_blank"
                      >
                        <span className="fa fa-external-link fa-fw mr-2" />
                        <span>
                          <I18n>Overhead</I18n>
                        </span>
                      </Link>
                    </DropdownMenu>
                  </Dropdown>
                )*/}
              </div>
              <Link
                to={`/appointment/${selectedTechBar.values['Id']}`}
                className="btn btn-primary card-button"
              >
                <I18n>Schedule Now</I18n> â†’
              </Link>
            </div>
          </div>
        </div>
      </div>
      <div className="page-container page-container--tech-bar container">
        <section>
          <div className="page-title">
            <div className="page-title__wrapper">
              <h3>
                <I18n>tech bar</I18n> /{' '}
              </h3>
              <h1>
                <I18n>Upcoming Appointments</I18n>
              </h1>
            </div>
          </div>
          <div className="cards__wrapper cards__wrapper--appt mb-3">
            {upcomingAppointments.map(appt => {
              const techBar = techBars.find(
                t => t.values['Id'] === appt.values['Scheduler Id'],
              );
              const date = moment.utc(appt.values['Event Date'], DATE_FORMAT);
              const start = moment.utc(appt.values['Event Time'], TIME_FORMAT);
              const end = start
                .clone()
                .add(appt.values['Event Duration'], 'minute');
              return (
                <Link
                  to={`/appointment/${appt.values['Scheduler Id']}/${appt.id}`}
                  className="card card--long card--appt"
                  key={appt.id}
                >
                  <i
                    className="fa fa-calendar fa-fw card-icon"
                    style={{ background: 'rgb(255, 74, 94)' }}
                  />
                  <div className="card-body">
                    <span className="card-title">
                      <Moment
                        timestamp={date}
                        format={Constants.MOMENT_FORMATS.dateWithDay}
                      />
                    </span>
                    <p className="card-subtitle">
                      <Moment
                        timestamp={start}
                        format={Constants.MOMENT_FORMATS.time}
                      />
                      {` - `}
                      <Moment
                        timestamp={end}
                        format={Constants.MOMENT_FORMATS.time}
                      />
                    </p>
                    {techBar && (
                      <p className="card-meta">
                        <strong>
                          <I18n>{techBar.values['Name']}</I18n>
                        </strong>
                      </p>
                    )}
                    <span
                      className={`badge ${
                        appt.coreState === 'Closed'
                          ? 'badge-dark'
                          : 'badge-success'
                      }`}
                    >
                      <I18n>{appt.values['Status']}</I18n>
                    </span>
                    <p className="card-text">{appt.values['Summary']}</p>
                  </div>
                </Link>
              );
            })}
            {upcomingAppointments.size === 0 &&
              !loadingUpcoming &&
              upcomingErrors.length === 0 && (
                <div className="text-center mx-auto text-muted">
                  <h3 className="mb-3">
                    <I18n>You have no upcoming appointments.</I18n>
                  </h3>
                  <p>
                    <I18n>
                      As you schedule appointments, they'll appear here.
                    </I18n>
                  </p>
                </div>
              )}
          </div>
          <Link to="/past" className="btn btn-link">
            <I18n>View Past Appointments</I18n>
          </Link>
        </section>
      </div>
      {modalOpen && (
        <Modal
          isOpen={!!modalOpen}
          toggle={() => setModalOpen(false)}
          size="sm"
          className="tech-bar-modal"
        >
          <div className="modal-header">
            <h4 className="modal-title">
              <button
                type="button"
                className="btn btn-link"
                onClick={() => setModalOpen(false)}
              >
                <I18n>Cancel</I18n>
              </button>
              <span>Location</span>
              <Link to="/tech-bars" className="btn btn-sm btn-link">
                <I18n>View All</I18n>
              </Link>
            </h4>
          </div>
          {!userLocation && (
            <div className="modal-header">
              <div className="px-3 py-1 text-center">
                <button className="btn btn-link" onClick={getUserLocation}>
                  <span className="fa fa-fw fa-map-marker" />{' '}
                  <I18n>use my current location</I18n>
                </button>
              </div>
            </div>
          )}
          <ModalBody>
            <ul>
              {techBars.map(techBar => (
                <li
                  key={techBar.id}
                  onClick={() => selectCurrentTechBar(techBar.id)}
                >
                  <span className="title">
                    <I18n>{techBar.values['Name']}</I18n>
                  </span>
                  {techBar.values['Location'] && (
                    <div className="subtitle">
                      <I18n>{techBar.values['Location']}</I18n>
                    </div>
                  )}
                  {userLocation && (
                    <div className="subtitle">
                      <I18n
                        render={translate =>
                          techBar.distance
                            ? `(${techBar.distance.toFixed(1)} ${translate(
                                'miles',
                              )})`
                            : `(${translate('Distance Unknown')})`
                        }
                      />
                    </div>
                  )}
                  <div className="details">
                    {techBar.values['Description'] && (
                      <p>
                        <I18n>{techBar.values['Description']}</I18n>
                      </p>
                    )}
                    {techBar.values['Details'] && (
                      <p>
                        <I18n>{techBar.values['Details']}</I18n>
                      </p>
                    )}
                  </div>
                </li>
              ))}
            </ul>
          </ModalBody>
        </Modal>
      )}
    </Fragment>
  );
};

export const mapStateToProps = (state, props) => {
  return {
    kapp: selectCurrentKapp(state),
    techBars: state.techBar.techBarApp.schedulers
      .filter(s => s.values['Status'] === 'Active')
      .map(
        props.userLocation
          ? mapTechBarsForDistance(props.userLocation)
          : t => t,
      )
      .sort(props.userLocation ? sortTechBarsByDistance : () => 0),
    loadingUpcoming: state.techBar.appointments.upcoming.loading,
    upcomingErrors: state.techBar.appointments.upcoming.errors,
    upcomingAppointments: state.techBar.appointments.upcoming.data,
    profile: state.app.profile,
  };
};

export const mapDispatchToProps = {
  push,
  fetchUpcomingAppointments: actions.fetchUpcomingAppointments,
};

const selectCurrentTechBar = ({
  techBars,
  setCurrentTechBar,
  setModalOpen,
}) => id => {
  setModalOpen(false);
  sessionStorage.setItem(SESSION_ITEM_CURRENT_TECH_BAR, id);
  setCurrentTechBar(techBars.find(t => t.id === id));
};

const getUserLocation = ({ setUserLocation }) => () => {
  navigator.geolocation.getCurrentPosition(position => {
    if (position && position.coords) {
      const userLocObj = {
        latitude: position.coords.latitude,
        longitude: position.coords.longitude,
      };
      setUserLocation(userLocObj);
      sessionStorage.setItem(
        SESSION_ITEM_USER_LOCATION,
        JSON.stringify(userLocObj),
      );
    }
  });
};

const toggleDropdown = ({
  setOpenDropdown,
  openDropdown,
}) => dropdownSlug => () =>
  setOpenDropdown(dropdownSlug === openDropdown ? false : dropdownSlug);

const hasTechBarDisplayRole = ({ profile }) => techBarName =>
  Utils.isMemberOf(profile, `Role::Tech Bar Display::${techBarName}`);

export const Home = compose(
  withState('userLocation', 'setUserLocation', () => {
    try {
      const locObj = JSON.parse(
        sessionStorage.getItem(SESSION_ITEM_USER_LOCATION),
      );
      if (locObj && locObj.latitude != null && locObj.longitude != null) {
        return locObj;
      }
    } catch (e) {}
    return null;
  }),
  connect(mapStateToProps, mapDispatchToProps),
  withState('currentTechBar', 'setCurrentTechBar', ({ techBars }) => {
    const techBarId = sessionStorage.getItem(SESSION_ITEM_CURRENT_TECH_BAR);
    return techBarId ? techBars.find(t => t.id === techBarId) : null;
  }),
  withState('modalOpen', 'setModalOpen', false),
  withState('openDropdown', 'setOpenDropdown', false),
  withHandlers({
    selectCurrentTechBar,
    getUserLocation,
    toggleDropdown,
    hasTechBarDisplayRole,
  }),
  lifecycle({
    componentDidMount() {
      if (!this.props.loadingUpcoming) {
        this.props.fetchUpcomingAppointments();
      }
      if (this.props.userLocation === null) {
        this.props.getUserLocation();
      }
    },
  }),
)(HomeComponent);
