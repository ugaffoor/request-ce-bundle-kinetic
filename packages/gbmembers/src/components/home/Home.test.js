import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Home } from './Home';

describe('<Home />', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Home />);
  });

  test('verify snapshot', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  test('should render a div containing a greeting', () => {
    const div = component.find('div').first();

    expect(div).toBeDefined();
    expect(div.text()).toEqual(expect.stringContaining('Hello'));
  });
});
