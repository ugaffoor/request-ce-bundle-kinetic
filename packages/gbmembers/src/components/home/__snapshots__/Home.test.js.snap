// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> verify snapshot 1`] = `
<SideEffect(DocumentTitle)
  title="Home"
>
  <div>
    Hello World!
  </div>
</SideEffect(DocumentTitle)>
`;
